hazelcast/src/main/java/com/hazelcast/config/IndexConfig.java

```diff
@@ -0,0 +1,254 @@
+     * @return This instance for chaining.
+     */
+    public IndexConfig setName(String name) {
+        this.name = name;
+
+        return this;
+    }
+
+    /**
+     * Gets type of the index.
+     * <p>
+     * Defaults to {@link IndexType#SORTED}.
+     *
+     * @return Type of the index.
+     */
+    public IndexType getType() {
+        return type;
+    }
+
+    /**
+     * Sets type of the index.
+     * <p>
+     * Defaults to {@link IndexType#SORTED}.
+     *
+     * @param type Type of the index.
+     * @return This instance for chaining.
+     */
+    public IndexConfig setType(IndexType type) {
+        this.type = checkNotNull(type, "Index type cannot be null.");
+
+        return this;
+    }
+
+    /**
+     * Gets index attributes.
+     *
+     * @return Index attributes.
+     */
+    public List<String> getAttributes() {
+        if (attributes == null) {
+            attributes = new ArrayList<>();
+        }
+
+        return attributes;
+    }
+
+    /**
+     * Adds an index attribute with the given.
+     *
+     * @param attribute Attribute name.
+     * @return This instance for chaining.
+     */
+    public IndexConfig addAttribute(String attribute) {
+        addAttributeInternal(attribute);
+
+        return this;
+    }
+
+    public void addAttributeInternal(String attribute) {
+        IndexUtils.validateAttribute(attribute);
+
+        if (attributes == null) {
+            attributes = new ArrayList<>();
+        }
+
+        attributes.add(attribute);
+    }
+
+    /**
+     * Sets index attributes.
+     *
+     * @param attributes Index attributes.
+     * @return This instance for chaining.
+     */
+    public IndexConfig setAttributes(List<String> attributes) {
+        checkNotNull(attributes, "Index attributes cannot be null.");
+
+        this.attributes = new ArrayList<>(attributes.size());
+
+        for (String attribute : attributes) {
+            addAttribute(attribute);
+        }
+
+        return this;
+    }
+
+    @Override
+    public int getFactoryId() {
+        return ConfigDataSerializerHook.F_ID;
+    }
+
+    @Override
+    public int getClassId() {
+        return ConfigDataSerializerHook.INDEX_CONFIG;
+    }
+
+    @Override
+    public void writeData(ObjectDataOutput out) throws IOException {
+        out.writeUTF(name);
+        out.writeInt(type.getId());
+        writeNullableList(attributes, out);
+    }
+
+    @Override
+    public void readData(ObjectDataInput in) throws IOException {
+        name = in.readUTF();
+        type = IndexType.getById(in.readInt());
+        attributes = readNullableList(in);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        IndexConfig that = (IndexConfig) o;
+
+        if (name != null ? !name.equals(that.name) : that.name != null) {
```

Comment:
`!Objects.equals(...)`?

Reply:
Fixed in https://github.com/hazelcast/hazelcast/pull/15537/commits/2163cbdd5ac5290a80317aa92679e8ded7520b42